### PR TITLE
style(suite): rework of trade page (menu)

### LINKED ISSUE #10557: Update the Trade page
I have left a bunch of comment in Figma with directions, hopefully that would provide more clarity! The designs might be updated soon if Ondra has time for it, let me know if any info is missing.

- [x] [Core section](https://www.figma.com/file/pewjlOfequOKYVC6h3v0YL/%5BEmpower-%231%5D-UI-Library-%26-Global-Navigation?type=design&node-id=8423-16902&mode=design&t=kjolSWNKovWkleuP-4)
- [x] [Offers section](https://www.figma.com/file/pewjlOfequOKYVC6h3v0YL/%5BEmpower-%231%5D-UI-Library-%26-Global-Navigation?type=design&node-id=3276-177456&mode=design&t=kjolSWNKovWkleuP-4)
- [x] [Checkout section](https://www.figma.com/file/pewjlOfequOKYVC6h3v0YL/%5BEmpower-%231%5D-UI-Library-%26-Global-Navigation?type=design&node-id=8424-16995&mode=design&t=kjolSWNKovWkleuP-4)
- [x] Move [this](https://github.com/trezor/trezor-suite/blob/334fcb95f64d748d1028861e45744335a68e1f87/packages/suite/src/views/wallet/coinmarket/savings/setup/index.tsx#L131) to `onClick` as it's not broken

### PR DESCRIPTION
**Pull request by peter-sanderson**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/10557

### CODE REVIEW

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
+import { useDevice, useSelector } from 'src/hooks/suite';
 import { getTitleForNetwork } from '@suite-common/wallet-utils';
 import { Translation } from 'src/components/suite';
 import { selectSelectedAccount } from 'src/reducers/wallet/selectedAccountReducer';
 import { FirmwareType } from '@trezor/connect';
+import { FONT_SIZE, FONT_WEIGHT } from '@trezor/components/src/config/variables';
+import { TranslationKey } from '@suite-common/intl-types';
+import { Route } from '@suite-common/suite-types';
+import { borders, spacingsPx } from '@trezor/theme';
+import { NavigationItem } from '../../../../../components/suite/Preloader/SuiteLayout/Sidebar/NavigationItem';
```
Comment (peter-sanderson) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:11]: this should be probably extraxted from `Preloader/SuiteLayout` to some shared place

File: packages/suite/src/components/suite/Preloader/SuiteLayout/Sidebar/NavigationItem.tsx [hunk 2]
```diff
@@ -68,16 +79,18 @@ export const NavigationItem = ({
...
-
+    const { elevation } = useElevation();
     const dispatch = useDispatch();
 
     const handleClick = (e: MouseEvent) => {
         e.stopPropagation();
 
         if (route) {
-            dispatch(goto(route));
+            dispatch(goto(route, preserveParams === true ? { preserveParams: true } : undefined));
```
Comment (jvaclavik) [inline packages/suite/src/components/suite/Preloader/SuiteLayout/Sidebar/NavigationItem.tsx:93]:
```
preserveParams === true ? { preserveParams } : undefined
```

WDYT?

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx [hunk 3]
```diff
@@ -1,25 +1,59 @@
...
+    gap: ${spacingsPx.xxs};
+
+    ${variables.SCREEN_QUERY.BELOW_LAPTOP} {
+        width: 100%;
+        flex-direction: row;
+        overflow: scroll;
+
+        ::-webkit-scrollbar {
+            display: none;
+        }
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:28]: I don't think it's a good idea to hide scrollbar here:

Screenshot 2024-01-25 at 13 02 42

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx [hunk 4]
```diff
@@ -1,25 +1,59 @@
...
-    margin-left: auto;
-    padding-left: 42px;
+    flex-direction: column;
+    width: 171px;
+    gap: ${spacingsPx.xxs};
+
+    ${variables.SCREEN_QUERY.BELOW_LAPTOP} {
+        width: 100%;
+        flex-direction: row;
+        overflow: scroll;
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:24]: Maybe `overflow-x: auto`?

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx [hunk 5]
```diff
@@ -1,25 +1,59 @@
...
-    { route: 'wallet-coinmarket-buy', title: 'TR_NAV_BUY' },
-    { route: 'wallet-coinmarket-sell', title: 'TR_NAV_SELL' },
-    { route: 'wallet-coinmarket-exchange', title: 'TR_NAV_EXCHANGE' },
-    { route: 'wallet-coinmarket-spend', title: 'TR_NAV_SPEND' },
-] as const;
-const p2pRoute = 'wallet-coinmarket-p2p';
+const NavListItem = styled(NavigationItem)`
+    white-space: nowrap;
+    font-size: ${FONT_SIZE.NORMAL};
+    font-weight: ${FONT_WEIGHT.MEDIUM};
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:35]:
```
${typography.body}
```

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx [hunk 6]
```diff
@@ -1,25 +1,59 @@
...
+            ? css`
+                  border-radius: ${borders.radii.full};
+              `
+            : ''}
+`;
+
+const Divider = styled.div`
+    margin: ${spacingsPx.xxs} 0; /* gap xxs + margin xxs = spacing xs around divider */
+    width: 100%;
+    border-bottom: 1px solid ${({ theme }) => theme.textDefault};
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:48]:
```jsx
border-bottom: 1px solid ${({ theme }) => theme.borderOnElevation0};
```

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx [hunk 7]
```diff
@@ -1,25 +1,59 @@
...
+
+    ${variables.SCREEN_QUERY.BELOW_LAPTOP} {
+        margin: 0 ${spacingsPx.xxs}; /* gap xxs + margin xxs = spacing xs around divider */
+        flex-direction: row;
+        border-bottom: 0;
+        border-right: 1px solid ${({ theme }) => theme.textDefault};
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:54]:
```jsx
border-right: 1px solid ${({ theme }) => theme.borderOnElevation0};

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketRefreshTime.tsx [hunk 8]
```diff
@@ -17,7 +24,7 @@ const RefreshLabel = styled.div`
 `;
 
 const RefreshTime = styled.div`
-    margin-left: 4px;
+    margin-left: ${spacingsPx.xxs};
     font-variant-numeric: tabular-nums;
     text-align: right;
     color: ${({ theme }) => theme.TYPE_DARK_GREY};
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketRefreshTime.tsx:30]:
```tsx
color: ${({ theme }) => theme.textSubdued};

**Comment by jvaclavik** [comment]
Few visual bugs:

1) 
Screenshot 2024-01-25 at 13 12 12

2) 
Screenshot 2024-01-25 at 13 13 57

3) 
Screenshot 2024-01-25 at 13 15 24

4)
Screenshot 2024-01-25 at 13 19 52

5) 
Screenshot 2024-01-25 at 13 18 40

6)
Screenshot 2024-01-25 at 13 17 17

File: packages/suite/src/components/suite/Preloader/SuiteLayout/Sidebar/NavigationItem.tsx (hunk 2 above)
Comment (peter-sanderson) [reply packages/suite/src/components/suite/Preloader/SuiteLayout/Sidebar/NavigationItem.tsx:93]: faca1f6098

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx (hunk 3 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:28]: f6dba6eaa2

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx (hunk 4 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:24]: 287ea89187

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx (hunk 5 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:35]: d0ad20c4d0

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx (hunk 6 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:48]: shall we do map for elevation and borders as well?

Comment (peter-sanderson) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:48]: e9cb7ffc4b

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx (hunk 7 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:54]: e9cb7ffc4b

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx (hunk 6 above)
Comment (jvaclavik) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketLayout/CoinmarketLayoutNavigation.tsx:48]: > shall we do map for elevation and borders as well?

Not today, but after merging to develop yes

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketRefreshTime.tsx (hunk 8 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketRefreshTime.tsx:30]: #ee1abcd67a

**Comment by dahaca** [comment]
Would be amazing to make the transactions section occupy the remaining space, with the footer actually being the footer. I can drop a commit once you are done, as I already have a solution :) 

Also here the tabs seem to have elevation0 and the border is strangely dark.

image

**Comment by peter-sanderson** [comment]
closed in favor of https://github.com/trezor/trezor-suite/pull/10784

**Pull request closed without merging**